biz.aQute.repository/src/aQute/bnd/repository/maven/util/RepoActionsUtil.java

```diff
@@ -21,42 +22,17 @@ public static void addSources(final Archive archive, Function<Archive, Promise<F
 
 		if (pBinary.getFailure() == null) {
 			final File binary = pBinary.getValue();
-			final File out = new File(binary.getParentFile(), "+" + binary.getName());
-
-			if (!out.isFile()) {
-				Archive a = archive.revision.archive("jar", "sources");
-				Promise<File> pSources = lookup.apply(a);
-
-				if (pSources.getFailure() == null) {
-					final File sources = pSources.getValue();
-
-					if (sources.isFile() && sources.length() > 1000) {
-						map.put("Add Sources", () -> {
-
-							try {
-
-								try (Jar src = new Jar(sources)) {
-
-									try (Jar bin = new Jar(binary)) {
-										bin.setDoNotTouchManifest();
-										for (String path : src.getResources()
-											.keySet())
-											bin.putResource("OSGI-OPT/src/" + path, src.getResource(path));
-										bin.write(out);
-									}
-									out.setLastModified(System.currentTimeMillis());
-								}
-							} catch (Exception e) {
-								throw Exceptions.duck(e);
-							}
-						});
-						return;
-					}
-				}
-			}
+
+			map.put("Add Sources", () -> {
+
+				Archive a = archive.getOther(JAR_EXTENSION, SOURCES_CLASSIFIER);
+				lookup.apply(a);
+
+			});
+			return;
 		}
 
-		map.put("-Add Sources", null);
+		// map.put("-Add Sources", null);
```

Comment:
Is this commented out line still necessary? If yes, a comment explaining its use would be helpful.

Reply:
I'll remove that too.